packages/ra-ui-materialui/src/list/pagination/Pagination.tsx

```diff
@@ -23,6 +23,8 @@ export const Pagination: FC<PaginationProps> = memo(props => {
         rowsPerPageOptions = DefaultRowsPerPageOptions,
         actions,
         limit = null,
+        hasNextPage: _hasNextPage,
+        hasPreviousPage: _hasPreviousPage,
```

Comment:
We should make the change in `useListController`'s `injectedProps` instead, the bug is earlier in the chain.

Reply:
Bump

Reply:
Bump

Reply:
Actually I can't reproduce the bug anymore. Reverting